cdk/lib/gatehouse.ts

```diff
@@ -0,0 +1,64 @@
+import {GuPlayApp} from '@guardian/cdk';
+import {AccessScope} from '@guardian/cdk/lib/constants/access';
+import {GuStack} from '@guardian/cdk/lib/constructs/core';
+import type {GuStackProps} from '@guardian/cdk/lib/constructs/core';
+import {GuCname} from '@guardian/cdk/lib/constructs/dns';
+import {GuPolicy, ReadParametersByName} from "@guardian/cdk/lib/constructs/iam";
+import type {App} from 'aws-cdk-lib';
+import {Duration} from 'aws-cdk-lib';
+import {InstanceClass, InstanceSize, InstanceType} from 'aws-cdk-lib/aws-ec2';
+
+export interface GatehouseStackProps extends GuStackProps {
+    domainName: string;
+}
+
+export class Gatehouse extends GuStack {
+    constructor(
+        scope: App,
+        id: string,
+        props: GatehouseStackProps,
+    ) {
+        super(scope, id, props);
+
+        const ec2App = 'gatehouse';
+
+        const readAppSsmParamsPolicy = new GuPolicy(this, 'ReadAppSsmParamsPolicy', {
+            statements: [new ReadParametersByName(this, {app: ec2App})]
+        })
+
+        const {loadBalancer} = new GuPlayApp(this, {
+            app: ec2App,
+            instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.MICRO),
+            access: {scope: AccessScope.PUBLIC},
+            userData: {
+                distributable: {
+                    fileName: `${ec2App}.deb`,
+                    executionStatement: `dpkg -i /${ec2App}/${ec2App}.deb\necho "stage=${this.stage}" | sudo tee "/etc/gatehouse/stage.conf" > /dev/null`,
```

Comment:
Is the `\n` character right here? It could probably be `;` or `&&` if the second command should only run once the first succeeds?

Reply:
I suppose if the first command fails the second is going to be irrelevant so I will take your advice and separate them with `&&`.

Reply:
Done in ddcabc59b18dec9908fce74cecd8566ca8ea69f6